qt/scientific_interfaces/Direct/ALFView.cpp

```diff
@@ -0,0 +1,41 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ALFView_model.h"
+#include "ALFView_presenter.h"
+
+// will need these later
+#include "MantidQtWidgets/Common/FunctionBrowser.h"
+#include "MantidQtWidgets/InstrumentView/InstrumentWidget.h"
+#include "MantidQtWidgets/InstrumentView/InstrumentWidgetPickTab.h"
+#include "MantidQtWidgets/Plotting/PreviewPlot.h"
+
+#include "ALFView.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+using namespace Mantid;
+
+DECLARE_SUBWINDOW(ALFView)
+
+/// static logger
+Mantid::Kernel::Logger g_log("ALFView");
+
+ALFView::ALFView(QWidget *parent)
+    : UserSubWindow(parent), m_view(nullptr), m_presenter(nullptr) {
+
+  m_view = new ALFView_view(this);
+  m_model = new ALFView_model();
```

Comment:
I think there is a memory leak with the model/presenter? I don't see them delete anywhere. Can we use `unique_ptr` for those as they are not in the Qt hierarchy so won't have a parent to delete them?

Reply:
I tried that and it give me stranger rendering issues. Would it be ok to instead add an explicit clean up to the header:
```
  ~ALFView() {
    delete m_presenter;
    delete m_model;
  };
```

Reply:
I think that would be fine in this case as Qt/smart pointer integration can get tricky.